test/unittests/test_extend_tt.cpp

```diff
@@ -10,52 +10,25 @@ class TruthTableExtend: public testing::Test {
 protected:
     TruthTable  tt{};
     std::string test_circuits_dir = "./circuits/";
-
-    TruthTable::Cube c00{};
-    TruthTable::Cube c11{};
-
-    TruthTable::Cube c000{};
-    TruthTable::Cube c001{};
-    TruthTable::Cube c010{};
-    TruthTable::Cube c011{};
-    TruthTable::Cube c100{};
-    TruthTable::Cube c101{};
-    TruthTable::Cube c110{};
-    TruthTable::Cube c111{};
-
-    void SetUp() override {
-        c00 = TruthTable::Cube::fromInteger(0b00U, 2U);
-        c11 = TruthTable::Cube::fromInteger(0b11U, 2U);
-
-        c000 = TruthTable::Cube::fromInteger(0b000U, 3U);
-        c001 = TruthTable::Cube::fromInteger(0b001U, 3U);
-        c010 = TruthTable::Cube::fromInteger(0b010U, 3U);
-        c011 = TruthTable::Cube::fromInteger(0b011U, 3U);
-        c100 = TruthTable::Cube::fromInteger(0b100U, 3U);
-        c101 = TruthTable::Cube::fromInteger(0b101U, 3U);
-        c110 = TruthTable::Cube::fromInteger(0b110U, 3U);
-        c111 = TruthTable::Cube::fromInteger(0b111U, 3U);
-    }
 };
 
 TEST_F(TruthTableExtend, Ident2Bit) {
     // create identity truth table
-
     std::string circIdent2Bit = test_circuits_dir + "Ident2Bit.pla";
 
     EXPECT_TRUE(read_pla(tt, circIdent2Bit));
 
-    EXPECT_EQ(tt.ioCube().size(), 3U);
+    EXPECT_EQ(tt.cubeMap.size(), 3U);
```

Comment:
the `cubeMap` should not be accessible from outside the class as previously pointed out.
use the public interface of the TT class instead.
This also holds for all the other test files.

If it is not already, make `cubeMap`a private member of the TT class.

Reply:
So I have a few questions on this. Initially, the cubeMap was private, however, you suggested that the extend, huffman  and augmentconstants function should not be in the truthtable class. These functions use the cubeMap member of the TT class. Since I separate them to different files, I had to make the cubeMap class public and TT class a struct. Therefore there is no public interface for the cubemap.

Reply:
What do you suggest I do?

Reply:
TruthTable should be a class and the cubeMap should be private (as should be all functions that need not be exposed publicly).

If the other functions need some sort of access to internal structures then provide respective public access/iteration methods in the truth table class.

Reply:
Okay, I will change this accordingly.

Reply:
done 👍🏼